src/ResourceManager/StorageSync/Commands.StorageSync/OutputWriters/TextSummaryOutputWriter.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace Microsoft.Azure.Commands.StorageSync.Evaluation.OutputWriters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Commands.StorageSync.Evaluation.Validations;
+    using Microsoft.Azure.Commands.StorageSync.Evaluation.Interfaces;
+
+    class TextSummaryOutputWriter : IOutputWriter
+    {
+        #region Fields and Properties
+        private readonly List<IValidationResult> _systemValidationResults;
+        private readonly Dictionary<ValidationType, long> _validationErrorsHistogram;
+        private readonly IConsoleWriter _consoleWriter;
+        private readonly HashSet<ValidationType> _systemValidationTypes;
+        private readonly Dictionary<ValidationType, string> _validationTypeDescriptions;
+
+        #endregion
+
+        #region Constructors
+        public TextSummaryOutputWriter(
+            IConsoleWriter consoleWriter, 
+            IList<IValidationDescription> validationDescriptions)
+        {
+            this._consoleWriter = consoleWriter;
+            this._validationErrorsHistogram = new Dictionary<ValidationType, long>();
+            this._systemValidationResults = new List<IValidationResult>();
+
+            this._systemValidationTypes = new HashSet<ValidationType>(validationDescriptions
+                .Where(o => o.ValidationKind == ValidationKind.SystemValidation)
+                .Select(o => o.ValidationType));
+
+            this._validationTypeDescriptions = new Dictionary<ValidationType, string>();
+            foreach (IValidationDescription validationDescription in validationDescriptions)
+            {
+                this._validationTypeDescriptions[validationDescription.ValidationType] = validationDescription.DisplayName;
+            }
+        }
+        #endregion
+
+        #region Public methods
+        public void Write(IValidationResult validationResult)
+        {
+            if (this.IsSystemValidation(validationResult))
+            {
+                this._systemValidationResults.Add(validationResult);
+            }
+            else
+            {
+                if (IsError(validationResult))
+                {
+                    if (!this._validationErrorsHistogram.ContainsKey(validationResult.Type))
+                    {
+                        this._validationErrorsHistogram[validationResult.Type] = 0;
+                    }
+
+                    this._validationErrorsHistogram[validationResult.Type] += 1;
+                }
+            }
+        }
+
+        public void WriteReport(string computerName, INamespaceInfo namespaceInfo)
+        {
+            if (this._systemValidationResults.Any())
+            {
+                this._consoleWriter.WriteLine(" ");
```

Comment:
@afedyashov we should avoid writing to the console in favor of one of the PowerShell streams so the user can access this information

Reply:
We intentionally made it go to console. In our design we are using multiple channels to provide information: validation results are going to output stream, that is designed to use in automation (e.g. act on errors found or export them to CSV), verbose channel will show verbose details of the scanning process including timing and configuration, progress bar gives feedback of the scanning process and finally console writer is specifically written to output to human readable summary information for customers convenience only (not for automation purposes), which can be silenced by the cmdline switch. It seems to us to make sense as we'll cover the needs of more sophisticated customers oriented on automation as well as less sophisticated, who'll find summary useful.

Reply:
It would be best if we used one of the provided streams to give this information to the user rather than using `Console.WriteLine` -- depending on what the information is, we can use the Debug or Warning streams. Using streams makes it easier for users to use built in common parameters to suppress the streams or reroute them to a specific variable that they can access later.